ur_robot_driver/src/dashboard_client_ros.cpp

```diff
@@ -224,15 +317,24 @@ bool DashboardClientROS::handleRunningQuery(const ur_dashboard_msgs::srv::IsProg
 bool DashboardClientROS::handleSavedQuery(ur_dashboard_msgs::srv::IsProgramSaved::Request::SharedPtr req,
                                           ur_dashboard_msgs::srv::IsProgramSaved::Response::SharedPtr resp)
 {
-  resp->answer = this->client_.sendAndReceive("isProgramSaved\n");
-  std::regex expected("(true|false) ([^\\s]+)");
-  std::smatch match;
-  resp->success = std::regex_match(resp->answer, match, expected);
+  try
```

Comment:
Maybe these almost the same block should be put in an external function in the future.  Just an idea.

Reply:
Since the msg response differs between the different calls this would not work. For all Trigger Services this was already done